### ISSUE TITLE
Cmake fails to fail for new TEST_GROUP() on Travis CI

### ISSUE THREAD

**Issue by arstrube** [cpputest/cpputest#638]
See here: https://travis-ci.org/arstrube/cpputest/builds/61118786

I added a test group `Utest` with a failing test to UtestTests.cpp in my fork, in order to figure out an unrelated problem with the AppVeyor build.

The Cmake build does not fail, apparently because the new test group is not included in the list of test groups, and thus not run.

**Comment by basvodde** [comment]
Oh excellent, less build failures :)

**Comment by arstrube** [comment]
hehe.

**Comment by arstrube** [comment]
@mmidor any idea?

**Comment by arstrube** [comment]
I created PR #640 with two tests that fail because of this. (The problem actually is that the name of one test group is _part_ of the name of another one.

The same thing can happen (though far less likely) with Group.name as well.

**Comment by arstrube** [comment]
```
TestRegistryTest.cpp:384: error: Failure in TEST(TestRegistry, shouldListTestNamesWhereOneIsContainedInAnother)
expected <MyTest.test HisTest.test Test.test>
but was <MyTest.test HisTest.test>
difference starts at position 24 at: <sTest.test >
^
TestRegistryTest.cpp:350: error: Failure in TEST(TestRegistry, shouldListGroupNamesWhereOneIsContainedInAnother)
expected <GROUP_1 GROUP>
but was <GROUP_1>
difference starts at position 7 at: < GROUP_1 >
^
```

**Comment by arstrube** [comment]
@mmidor -- is there actually even a chance that the same group or group.test occur more than once in the TestRegistry? If not, you could simply omit the test for contains().

If there is the chance, might it not be better to run the same group more than once, rather then omitting it altogether?

**Comment by Andne** [comment]
Test groups can cause some trouble for junit I know, because if a group isn't contiguous, then the file gets overwritten.  However, I think that for a given group.name, there's no way that you could ever have a duplicate without causing a linker error.

A while ago there was discussion over breaking out the test groups into an actual class, which would help junit I know and might help this.  If the test group is a class that then executes its tests, we would know that a group will always be executed as a single block instead of intermixed with other tests from other groups.  I suspect it would also mean that a group can only be registered to test registry once since otherwise it would cause an error with a type mismatch.  Granted, namespaces may cause some unexpected corner cases with that.

**Comment by arstrube** [comment]
Hmm. We know that any given group.name will be unique. But I am actually not so sure what happens if you define a test group in two compilation units, perhaps each with different test names. So maybe we need to play safe, after all.

**Comment by mmidor** [comment]
Yes, I made this change on a later stage because I noticed I have groups listed more than once.
I don't remember details now...
But your observation about contains() is correct - this conditional should be corrected.
Now, if group named Group1 was registered first, the Group12 might be omitted.

**Comment by arstrube** [comment]
I'll take your word for it. Any idea on your side how we can deal with the "containment" issue?

**Comment by mmidor** [comment]
Please, take a look at #642

**Comment by arstrube** [comment]
Neat. Thanks :-)

**Comment by mmidor** [comment]
Your welcome :) Sorry for delay, I didn't realized quickly what is all about

**Comment by arstrube** [comment]
This is fixed as #642 has been merged.
